pkg/agent/manager/cache/cache.go

```diff
@@ -386,20 +393,77 @@ func (c *Cache) UpdateSVIDs(update *UpdateSVIDs) {
 		}
 
 		record.svid = svid
-		notifySet.Merge(record.entry.Selectors...)
 		log := c.log.WithFields(logrus.Fields{
 			telemetry.Entry:    record.entry.EntryId,
 			telemetry.SPIFFEID: record.entry.SpiffeId,
 		})
 		log.Debug("SVID updated")
 
+		// TODO: may we allow subscribe to notifications when SVID is exportable?
+		// at the same time may we add a filter to avoid using this entries for attestation?
+		if record.entry.StoreSvid {
+			if err := c.storeSVID(record); err != nil {
+				log.WithError(err).Error("failed to export SVID")
+			}
+		} else {
+			notifySet.Merge(record.entry.Selectors...)
+		}
+
 		// Registration entry is updated, remove it from stale map
 		delete(c.staleEntries, entryID)
 	}
 
 	c.notifyBySelectors(notifySet)
 }
 
+func (c *Cache) storeSVID(record *cacheRecord) error {
+	plugName, err := getPluginName(record.entry.Selectors)
+	if err != nil {
+		return err
+	}
+
+	pipeIn, ok := c.pipesIn[plugName]
+	if !ok {
+		return fmt.Errorf("no SVID Store %q found", plugName)
+	}
+	update := &pipe.SVIDUpdate{
+		Entry:            record.entry,
+		SVID:             record.svid.Chain,
+		PrivateKey:       record.svid.PrivateKey,
+		Bundle:           c.bundles[c.trustDomainID],
+		FederatedBundles: make(map[string]*bundleutil.Bundle),
+	}
+
+	for _, federatesWith := range record.entry.FederatesWith {
+		if federatedBundle := c.bundles[federatesWith]; federatedBundle != nil {
+			update.FederatedBundles[federatesWith] = federatedBundle
+		} else {
+			c.log.WithFields(logrus.Fields{
+				telemetry.Entry:           record.entry.EntryId,
+				telemetry.SPIFFEID:        record.entry.SpiffeId,
+				telemetry.FederatedBundle: federatesWith,
+			}).Warn("Federated bundle contents missing")
+		}
+	}
+
+	pipeIn.Push(update)
+	return nil
+}
+
+func getPluginName(selectors []*common.Selector) (string, error) {
+	if len(selectors) == 0 {
+		return "", errors.New("no selectors found")
+	}
+
+	name := selectors[0].Type
+	for _, s := range selectors {
+		if name != s.Type {
+			return "", errors.New("selector contains multiple types")
```

Comment:
I think that it would be good to include in the error the expected type and the type found.

Reply:
mmm not sure about that because the guy that is creating it will contain that list, and we are not collecting all types we just fails at first one that is different.
if another are ok about returning that as part of error, are you ok about refactor this code to generate a map with all types and return is as part of error when len(map) > 1 ?